tests/config_input/template.yaml

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2012-2022 by the GalSim developers team on GitHub
+# https://github.com/GalSim-developers
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+
+
+template: multirng
+
+modules: 
+  - template_register
+  - astropy.time
+
+output:
+  file_name: test_template.fits
+
+gal.shear:
+  type: E1E2
+  e1: 0.2
+  e2: 0
+
+input.power_spectrum.1.grid_spacing: 2
+
+gal.magnify:
+  type: PowerSpectrumMagnification
+  num: 1
+
+gal.ellip:
+  template: multirng:psf.ellip
```

Comment:
This is good, but just a note that it doesn't explicitly test the thing I noted:

```
file1.yaml
  template: mytemplate.yaml

file2.yaml
  template: file1.yaml
```

Although from reading the code I think it is handled the same.